app/frontend/src/registry/components/people/ApplicationAdd.vue

```diff
@@ -1,218 +1,156 @@
-      const quals = []
-      const activity = this.qualCodes.find(item => item.activity === this.qualType)
-      return activity ? activity.quals : quals
+  watch: {
+    // Watching the entire object is expensive, but we need some way of notifying the parent.
+    computedQualificationForm: {
+      handler: function (val, oldVal) {
+        this.$emit('input', val)
+      },
+      deep: true
     }
   },
-  watch: {
-    qualType: function () {
-      const qualList = this.quals.map(item => item.toLowerCase())
-      for (let key in this.editClassification.qualCodes) {
-        this.editClassification.qualCodes[key] = false
+  methods: {
+    changedClassification (value) {
+      const match = this.subactivityMap.filter(item => item.registries_subactivity_code === value)[0]
+      this.qualificationForm.qualifications = match.qualification_set.map(item => item.well_class)
+    },
+    ...mapActions([
+      'fetchDrillerOptions'
+    ])
+  },
+  computed: {
+    computedQualificationForm: function () {
+      // We need to transform the bound dates to something that is acceptable to the API without affecting
+      // the values to which the form are bound.
+      // Make a deep copy, and transform the date fields.
+      const transformed = JSON.parse(JSON.stringify(this.qualificationForm))
+      transformed.status_set.forEach((status) => { status.effective_date = status.effective_date && status.effective_date.length >= 10 ? status.effective_date.substring(0, 10) : status.effective_date })
+      return transformed
+    },
+    ...mapGetters([
+      'loading'
+    ]),
+    formOptions () {
+      let result = {
+        issuer: [{value: null, text: 'Please select an option'}],
+        classifications: [],
+        qualifications: []
       }
-      for (let i = 0; i < qualList.length; i++) {
-        if (qualList[i] in this.editClassification.qualCodes) {
-          this.editClassification.qualCodes[qualList[i]] = true
-        }
+      if (this.$options.propsData.activity in this.$store.getters.drillerOptions) {
+        const options = this.$store.getters.drillerOptions[this.$options.propsData.activity]
+        result.issuer = result.issuer.concat(options.AccreditedCertificateCode.map((item) => { return {'text': item.name + ' (' + item.cert_auth + ')', 'value': item.acc_cert_guid} }))
+        result.classifications = options.SubactivityCode.map((item) => { return {'text': item.description, 'value': item.registries_subactivity_code} })
+        result.qualifications = options.WellClassCode.map((item) => { return {'text': item.description, 'value': item.registries_well_class_code} })
       }
+      return result
+    },
+    subactivityMap () {
+      return this.$options.propsData.activity in this.$store.getters.drillerOptions ? this.$store.getters.drillerOptions[this.$options.propsData.activity].SubactivityCode : []
     }
+  },
+  created () {
+    this.fetchDrillerOptions({activity: this.$options.propsData.activity})
```

Comment:
This component already has some declared properties `props: ['value', 'activity']` so you can access them with `this.activity` and `this.value`